x-pack/plugins/security_solution/public/timelines/pages/timelines_page.tsx

```diff
@@ -42,63 +40,76 @@ export const TimelinesPageComponent: React.FC = () => {
   const capabilitiesCanUserCRUD: boolean =
     !!useKibana().services.application.capabilities.siem.crud;
 
+  const createNewTimeline = useCreateTimeline({
+    timelineId: TimelineId.active,
+    timelineType: TimelineType.default,
+  });
+  const createNewTimelineTemplate = useCreateTimeline({
+    timelineId: TimelineId.active,
+    timelineType: TimelineType.template,
+  });
```

Comment:
This hook is called twice, but only one of the values returned is used, which makes me think that maybe it would make it cleaner if we extracted this logic in a separate `NewTimelineButton` component in this same file, like:

```
const NewTimelineButton = React.memo(({ type = TimelineType.default }: { type?: TimelineType }) => {
  const createNewTimeline = useCreateTimeline({
    timelineId: TimelineId.active,
    timelineType: type,
  });

  const dataTestSubj = `timelines-page-create-new-${
    type === TimelineType.default ? 'timeline' : 'timeline-template'
  }`;

  return (
    <EuiButton
      iconType="plusInCircle"
      data-test-subj={dataTestSubj}
      onClick={() => createNewTimeline()}
      fill
    >
      {type === TimelineType.default ? i18n.NEW_TIMELINE : i18n.NEW_TEMPLATE_TIMELINE}
    </EuiButton>
  );
});
```

(If it can be used in other places we could also make it generic and move it to a common place.)

Reply:
good idea, I extracted that logic into a component. It makes the testing of the `timelines_page` component a bit more challenging as we're using the `shallow` function from `enzyme` so this is now not rendered anymore...
I added unit tests for this new component so we're in good shape I think.